epochStart/metachain/auctionListSorting.go

```diff
@@ -0,0 +1,104 @@
+package metachain
+
+import (
+	"bytes"
+	"math/big"
+	"sort"
+
+	"github.com/ElrondNetwork/elrond-go/state"
+)
+
+func (als *auctionListSelector) selectNodes(
+	ownersData map[string]*ownerData,
+	numAvailableSlots uint32,
+	randomness []byte,
+) []state.ValidatorInfoHandler {
+	selectedFromAuction := make([]state.ValidatorInfoHandler, 0)
+	validatorTopUpMap := make(map[string]*big.Int)
+
+	pubKeyLen := getPubKeyLen(ownersData)
+	normRand := calcNormalizedRandomness(randomness, pubKeyLen)
+
+	for _, owner := range ownersData {
+		sortListByPubKey(owner.auctionList)
+		addQualifiedValidatorsTopUpInMap(owner, validatorTopUpMap)
+		selectedFromAuction = append(selectedFromAuction, owner.auctionList[:owner.numQualifiedAuctionNodes]...)
+	}
+
+	als.displayOwnersSelectedNodes(ownersData)
+	sortValidators(selectedFromAuction, validatorTopUpMap, normRand)
+	als.displayAuctionList(selectedFromAuction, ownersData, numAvailableSlots)
+
+	return selectedFromAuction[:numAvailableSlots]
+}
+
+func getPubKeyLen(ownersData map[string]*ownerData) int {
```

Comment:
what is this ? you do not need to iterate and return the lenght of data like this.

Reply:
This does not iterate over all values; it stops at the first element; I need it to find the blsKeyLen so we can normalize randomness